src/deepsparse/server/build_logger.py

```diff
@@ -0,0 +1,146 @@
+from deepsparse.server.config import ServerConfig
+
+
+__all__ = ["build_logger"]
+
+_LOGGER_MAPPING = {"python": PythonLogger}
+
+
+def build_logger(server_config: ServerConfig) -> BaseLogger:
+    """
+    Builds a DeepSparse logger from the ServerConfig.
+
+    The process follows the following hierarchy:
+
+    First: if global logger config is provided,
+        the "leaf" loggers are built.
+
+    Second: if data logging config is specified, a set of
+        function loggers wraps around the appropriate "leaf" loggers.
+
+    Third: The resulting loggers are wrapped inside a MultiLogger.
+
+    :param server_config: the Server configuration model
+    :return: a DeepSparse logger instance
+    """
+
+    loggers_config = server_config.loggers
+    if not loggers_config:
+        return None
+
+    leaf_loggers = build_leaf_loggers(loggers_config)
+    loggers = build_function_loggers(server_config.endpoints, leaf_loggers)
+    return MultiLogger(loggers)
+
+
+def build_leaf_loggers(
+    loggers_config: Dict[str, Optional[Dict[str, Any]]]
+) -> Dict[str, BaseLogger]:
+    """
+    Instantiate a set of leaf loggers according to the configuration
+
+    :param loggers_config: Config; specifies the leaf loggers to be instantiated
+    :return: A dictionary that contains a mapping from a logger's name to its instance
+    """
+    loggers = {}
+    for logger_name, logger_arguments in loggers_config.items():
+
+        logger_to_instantiate = _LOGGER_MAPPING.get(logger_name)
+        if logger_to_instantiate is None:
+            raise ValueError(
+                f"Unknown logger name: {logger_name}. "
+                f"supported logger names: {list(_LOGGER_MAPPING.keys())}"
+            )
+        logger_arguments = {} if logger_arguments is None else logger_arguments
+        leaf_logger = logger_to_instantiate(**logger_arguments)
+        loggers.update({logger_name: leaf_logger})
+    return loggers
+
+
+def build_function_loggers(
+    endpoints: List["EndpointConfig"], loggers: Dict[str, BaseLogger]  # noqa F821
+) -> List[FunctionLogger]:
+    """
+    Build a set of function loggers according to the configuration.
+
+    :param endpoints: A list of server's endpoint configurations;
+        the configurations contain the information about the metric
+        functions (MetricFunctionConfig objects)
+        and the targets that the functions are to be applied to
+    :param loggers: The created "leaf" loggers
+    :return: A list of FunctionLogger instances
+    """
+    function_loggers = []
+    for endpoint in endpoints:
+        if endpoint.data_logging is None:
+            continue
+        for target_logging_dict in endpoint.data_logging:
+            target = target_logging_dict["target"]
+            target_identifier = _get_target_identifier(
+                endpoint.name or endpoint.task, target
+            )
+            for metric_function_cfg in target_logging_dict["functions"]:
```

Comment:
this is the piece where I'm missing the `Union[..., str]` case.
quick suggestion:
```suggestion
        for target, metric_function_cfgs in endpoint.data_logging.items():
            target_identifier = _get_target_identifier(
                endpoint.name or endpoint.task, target
            )
            for metric_function_cfg in metric_function_cfgs:
```